cpp/src/lists/contains.cu

```diff
@@ -0,0 +1,252 @@
+/*
+ * Copyright (c) 2020, NVIDIA CORPORATION.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include <thrust/iterator/constant_iterator.h>
+#include <thrust/logical.h>
+#include <cudf/column/column_factories.hpp>
+#include <cudf/detail/iterator.cuh>
+#include <cudf/detail/valid_if.cuh>
+#include <cudf/lists/contains.hpp>
+#include <cudf/lists/list_device_view.cuh>
+#include <cudf/lists/lists_column_device_view.cuh>
+#include <cudf/lists/lists_column_view.hpp>
+#include <cudf/scalar/scalar.hpp>
+#include <cudf/scalar/scalar_device_view.cuh>
+#include <cudf/utilities/type_dispatcher.hpp>
+#include <rmm/exec_policy.hpp>
+#include <type_traits>
+
+namespace cudf {
+namespace lists {
+
+namespace {
+
+auto get_search_keys_device_iterable_view(cudf::column_view const& search_keys,
+                                          rmm::cuda_stream_view stream)
+{
+  return column_device_view::create(search_keys, stream);
+}
+
+auto get_search_keys_device_iterable_view(cudf::scalar const& search_key,
+                                          rmm::cuda_stream_view stream)
```

Comment:
I think this unused parameter might cause compiler warnings in some cases. You can make it an anonymous parameter...

```suggestion
                                          rmm::cuda_stream_view)
```

Reply:
Thank, @harrism. In the same vein, I have removed the name of the unused template bool parameter in `get_pair_iterator()`.